flink-java/src/main/java/org/apache/flink/api/java/DataSet.java

```diff
@@ -1727,6 +1727,21 @@ public void printToErr() throws Exception {
 		return output(new PrintingOutputFormat<T>(sinkIdentifier, true));
 	}
 
+	/**
+	 * Writes a DataSet using a {@link FileOutputFormat} to a specified location.
+	 * This method adds a data sink to the program.
+	 *
+	 * @param outputFormat The FileOutputFormat to write the DataSet.
+	 * @return The DataSink that writes the DataSet.
+	 *
+	 * @see FileOutputFormat
+	 */
+	public DataSink<T> write(FileOutputFormat<T> outputFormat) {
+		Preconditions.checkNotNull(outputFormat, "Output format must not be null.");
+		Preconditions.checkNotNull(outputFormat.getOutputFilePath(), "File path must not be null.");
+		return output(outputFormat);
```

Comment:
this right here is already a viable alternative for users, hence I would reject this PR.